datasetinsights/stats/visualization/coco_data_visualization.py

```diff
@@ -0,0 +1,78 @@
+import random
+from typing import List, Union
+
+import matplotlib.pyplot as plt
+from pycocotools.coco import COCO
+
+from datasetinsights.stats.coco_statistics import (
+    load_annotations_for_single_img,
+    load_image_from_img_ann,
+    load_img_ann_for_single_image,
+)
+
+
+def display_single_img(
+    data_dir: str,
+    img_id: int,
+    annotation_file: str = None,
+    show_annotation=False,
+) -> plt.Figure:
+    """
+    Displays single image with the give image id
+    Args:
+        annotation_file (str): COCO annotation json file.
+        img_id (int): Image id of the image
+        data_dir (str): Directory where data(images) is located
+        show_annotation (bool): Show annotation for the image
+
+    Returns:
+        plt.Figure: Figure object
+
+    """
+    coco = COCO(annotation_file=annotation_file)
```

Comment:
This should be avoided in a function that's generic like this. Using COCO instances couple this method to datasets in COCO format.

Reply:
This file is specifically for coco-like datasets and visualization. Do we need to make it more generic for different datasets. Also the logic with loading and visualizing coco annotation might be different from other datasets.